### ISSUE TITLE
Passing field data to header/footer

### ISSUE THREAD

**Issue by compass360** [enupal/snapshot#52]
Is it possible to pass the text from an entry field into header template? The documentation on variables is lacking, but I'm not sure if headers/footers are treated differently anyway.

I see there is some built-in variable-replacement functionality (title/section/pagination), but I'd like to be able to pass a simple string value from a field (entry.fieldName) so that it can appear in my headers across all pages.

Can this be done? Even if it requires JavaScript?

**Comment by andrelopez** [comment]
Hi @compass360 Yes, you can pass variables to templates and headers, for example if you're following the [headers](https://docs.enupal.com/enupal-snapshot/pdf-examples/footers-and-headers.html#adding-the-footer-and-header-to-the-clioptions) example you can pass variables using the `variables` key, pass the `entry` variable that will be avialable on the `pdfexamples/header` and `_pdfexamples/footer` templates: 

```twig
{%  set settings = {
        filename: 'my-first-pdf.pdf',
        variables: {
	    entry: entry
	},
        cliOptions: {
        	'header-html': '_pdfexamples/header',
        	'footer-html': '_pdfexamples/footer'
      	}
    }
%}

{{ craft.enupalsnapshot.displayHtml(html, settings) }}
```

**Comment by compass360** [comment]
How does the variable get exposed to the template? Using your example, I'm passing the variable in the settings as you've described, then trying {{ entry.title }} in the body of my header template. I get a Twig error (Variable "entry" does not exist). Similarly, "entry" does not appear to be a part of the query strings variable object used by the javascript in these templates... So how do I access the variables?

**Comment by compass360** [comment]
For sure! I'm on Snapshot 1.2.8 and Craft CMS 3.7.41. To troubleshoot, I've simplified the variable to simply try and get a text string (rather than an element/object) and getting the same result. The rest of the headers and footers appear to be outputting just fine, as outlined in your documentation.

`{%  set settings = {
  filename: '{{ entry.slug }}_{{ "now"|date("m-d-Y") }}.pdf',
  variables: {
    entryTitle: 'Sample Title'
  },
  cliOptions: {
    'page-size': 'Letter',
    'header-html': '/backend/generate-pdf-header.twig',
    'footer-html': '/backend/generate-pdf-footer.twig'
	}
} %}`

**Comment by andrelopez** [comment]
Hi @compass360 We just released Enupal Snapshot v1.2.9 with a fix to this issue. Thanks for reporting
